### PR TITLE
build: use ssh cmd's exit code rather than mkdir's

### PR DESCRIPTION
**Pull request by dt**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/13985)

### CODE REVIEW

**Review by jordanlewis — APPROVED** [review]

File: build/teamcity-jepsen.sh [hunk 1]
```diff
@@ -61,13 +61,15 @@ for test in "${tests[@]}"; do
...
-        # If this begins to time out frequently, let's do this via nohup and poll.
-        ssh -o "ServerAliveInterval=60" -o "StrictHostKeyChecking no" -i "$HOME/.ssh/${KEY_NAME}" "ubuntu@${controller}" "${testcmd}"
+
         # Remove spaces from test name to get the artifacts subdirectory
         testname=$(echo "${test}${nemesis}" | sed 's/ //g')
         artifacts_dir="${LOG_DIR}/${testname}"
         mkdir -p "${artifacts_dir}"
+
+        # Run each test over an ssh connection.
+        # If this begins to time out frequently, let's do this via nohup and poll.
```
Comment (tamird) [inline build/teamcity-jepsen.sh]: This is subtle and at least needs a comment, but also I don't love that this script doesn't use `set -e`, apparently?

**Review by tamird — APPROVED** [review]

File: build/teamcity-jepsen.sh (hunk 1 above)
Comment (dt) [inline build/teamcity-jepsen.sh]: there's already a comment about `set -e`.

`$? -ne 0` immediately after cmd is a common bash idiom, so don't think we need to comment about that -- this bug could happen with or without a comment, and is more of an argument against bash than anything else.

Comment (tamird) [inline build/teamcity-jepsen.sh]: urgh, that's true. btw, why is `mkdir` failing?

```shell
$ mkdir foo
$ mkdir -p foo
$ echo $?
0
```

Comment (dt) [inline build/teamcity-jepsen.sh]: it _isn't_ failing, that's the problem.

Comment (tamird) [inline build/teamcity-jepsen.sh]: :+1:

**Comment by dt** [comment]
If anyone gets bored, we can port this to a Go test, since bash is unmaintainable and terrible, but it does the job for now.

**Pull request merged**